python/plugins/processing/algs/gdal/GdalAlgorithm.py

```diff
@@ -130,6 +130,11 @@ def getOgrCompatibleSource(self, parameter_name, parameters, context, feedback,
             ogr_layer_name = GdalUtils.ogrLayerName(input_layer.dataProvider().dataSourceUri())
         return ogr_data_path, ogr_layer_name
 
```

Comment:
You can avoid this by storing the layer's path as a member in the algorithm during the processAlgorithm step, and then retrieving it in the postProcess step